myst_nb/parser.py

```diff
@@ -21,6 +27,94 @@ class NotebookParser(MystParser):
     config_section_dependencies = ("parsers",)
 
     def parse(self, inputstring, document):
+
+        # de-serialize the notebook
+        ntbk = nbf.reads(inputstring, nbf.NO_CONVERT)
+
+        # This is a contaner for top level markdown tokens
+        # which we will add to as we walk the document
+        mkdown_tokens = []
+
+        # first we ensure that we are using a 'clean' global context
+        # for parsing, which is setup with the MyST parsing tokens
+        parse_context = ParseContext(
```

Comment:
Is ParseContext a new addition to mistletoe from us?

Reply:
Yes, it is a consolidation of a number of aspects of the original code that utilised global variables. Now we contain them in a single place, that is also thread-safe to use in multithreaded applications (e.g. for LSP) 😄